### PR TITLE
feat: Added report to show Diagnosis trends

### PR DESCRIPTION
**Pull request by ChillarAnand**
Screenshot 2021-09-27 at 15 12 00

Screenshot 2021-09-27 at 15 12 09

### CODE REVIEW

File: healthcare/healthcare/report/diagnosis_trends/diagnosis_trends.js [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+					entry = {
+						'name': row_name,
+						'values': row_values
+					}
+
+					let raw_data = frappe.query_report.chart.data;
+					let new_datasets = raw_data.datasets;
+
+					let found = false;
+					for (let i=0; i < new_datasets.length;i++) {
```
Comment (ruchamahabal) [inline healthcare/healthcare/report/diagnosis_trends/diagnosis_trends.js]:
```suggestion
					for (let i=0; i < new_datasets.length; i++) {
```

File: healthcare/healthcare/report/diagnosis_trends/diagnosis_trends.py [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+import frappe
+from frappe import _, scrub
+from frappe.utils import add_days, add_to_date, flt, getdate
+from six import iteritems
+
+from erpnext.accounts.utils import get_fiscal_year
+
+
+def execute(filters=None):
+	return DiagnosisTrends(filters).run()
```
Comment (ruchamahabal) [inline healthcare/healthcare/report/diagnosis_trends/diagnosis_trends.py:14]: Can we add tests to check if correct data is being fetched based on Department, Dates, and Range filters?

Comment (ChillarAnand) [reply healthcare/healthcare/report/diagnosis_trends/diagnosis_trends.py:14]: Sure @ruchamahabal

File: healthcare/healthcare/report/diagnosis_trends/diagnosis_trends.json [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+ "modified": "2021-09-08 11:26:17.566266",
+ "modified_by": "Administrator",
+ "module": "Healthcare",
+ "name": "Diagnosis Trends",
+ "owner": "Administrator",
+ "prepared_report": 0,
+ "ref_doctype": "Patient Encounter Diagnosis",
+ "report_name": "Diagnosis Trends",
+ "report_type": "Script Report",
+ "roles": []
```
Comment (ruchamahabal) [inline healthcare/healthcare/report/diagnosis_trends/diagnosis_trends.json]: Add default roles for this report

File: healthcare/healthcare/report/diagnosis_trends/diagnosis_trends.py [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+
+		if department:
+			encounters = frappe.get_all('Patient Encounter', filters={'medical_department': department}, pluck='name')
+			if encounters:
+				_operator = OPERATOR_MAP['in']
+				query = query.where(_operator('parent', encounters))
+
+		self.entries = query.run(as_dict=True)
+		print(self.entries)
+		print(query.get_sql())
```
Comment (ruchamahabal) [inline healthcare/healthcare/report/diagnosis_trends/diagnosis_trends.py]:
```suggestion
```

File: healthcare/healthcare/report/diagnosis_trends/diagnosis_trends.json [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+ "idx": 0,
+ "is_standard": "Yes",
+ "modified": "2021-09-08 11:26:17.566266",
+ "modified_by": "Administrator",
+ "module": "Healthcare",
+ "name": "Diagnosis Trends",
+ "owner": "Administrator",
+ "prepared_report": 0,
+ "ref_doctype": "Patient Encounter Diagnosis",
+ "report_name": "Diagnosis Trends",
```
Comment (ruchamahabal) [inline healthcare/healthcare/report/diagnosis_trends/diagnosis_trends.json:19]: Add this report to the workspace?

**Review by ruchamahabal — CHANGES_REQUESTED** [review]

File: healthcare/healthcare/report/diagnosis_trends/diagnosis_trends.js (hunk 1 above)
Comment (ChillarAnand) [reply healthcare/healthcare/report/diagnosis_trends/diagnosis_trends.js]: Will setup sider to catch these issues.

**Review by ruchamahabal — APPROVED** [review]

**Pull request merged**